src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+    public List<Car> getAllByDriver(Long driverID) {
+        String getAllByDriverQuery = "SELECT cars.id AS car_id FROM cars_drivers "
+                + "JOIN cars ON cars_drivers.cars_id = cars.id WHERE drivers_id = ?";
+        List<Long> carsIDs = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement = connection
+                        .prepareStatement(getAllByDriverQuery)) {
+            getAllByDriverStatement.setLong(1, driverID);
+            ResultSet carsListFromDB = getAllByDriverStatement.executeQuery();
+            while (carsListFromDB.next()) {
+                carsIDs.add(carsListFromDB.getObject("car_id", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't select all cars which are belong"
+                    + " to driver with current ID: " + driverID, e);
+        }
+        List<Car> cars = new ArrayList<>();
+        for (Long carID : carsIDs) {
+            cars.add(get(carID).get());
+        }
+        return cars;
+    }
+
+    private void insertDrivers(Car car) {
+        String insertDriversQuery = "INSERT INTO cars_drivers (cars_id, drivers_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertDriversStatement = connection
+                        .prepareStatement(insertDriversQuery)) {
+            insertDriversStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertDriversStatement.setLong(2, driver.getId());
+                insertDriversStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert Drivers from input data: " + car, e);
+        }
+    }
+
+    private Car parseCarByDataFromDB(ResultSet dataFromDB) {
+        try {
+            Manufacturer currentManufacturer = new Manufacturer(dataFromDB
+                    .getString("manufacturer_name"),
+                    dataFromDB.getString("manufacturer_country"));
+            currentManufacturer.setId(dataFromDB.getObject("manufacturer_id", Long.class));
+
+            Car currentCar = new Car(dataFromDB.getString("model"), currentManufacturer);
+            currentCar.setId(dataFromDB.getObject("car_id", Long.class));
+            return currentCar;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't parse Car object from input data: "
+                    + dataFromDB, e);
+        }
+    }
+
+    private List<Driver> getDriversForCurrentCar(Long carID) {
+        String getDriversQuery = "SELECT drivers.id AS driver_id, drivers.name AS driver_name, "
+                + "drivers.license_number AS driver_license_number FROM drivers JOIN cars_drivers "
+                + "ON drivers.id = cars_drivers.drivers_id WHERE cars_id = ?;";
```

Comment:
let's add where condition so we won't fetch deleted drivers

Reply:
added